extensions/community/AudioByName.json

```
@@ -0,0 +1,202 @@
+{
+  "author": "",
+  "category": "Audio",
+  "extensionNamespace": "",
+  "fullName": "Audio by Filename",
+  "helpPath": "",
+  "iconUrl": "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4NCjwhLS0gR2VuZXJhdG9yOiBBZG9iZSBJbGx1c3RyYXRvciAyMy4wLjMsIFNWRyBFeHBvcnQgUGx1Zy1JbiAuIFNWRyBWZXJzaW9uOiA2LjAwIEJ1aWxkIDApICAtLT4NCjxzdmcgdmVyc2lvbj0iMS4xIiBpZD0iSWNvbnMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4Ig0KCSB2aWV3Qm94PSIwIDAgMzIgMzIiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDMyIDMyOyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+DQo8c3R5bGUgdHlwZT0idGV4dC9jc3MiPg0KCS5zdDB7ZmlsbDpub25lO3N0cm9rZTojMDAwMDAwO3N0cm9rZS13aWR0aDoyO3N0cm9rZS1saW5lam9pbjpyb3VuZDtzdHJva2UtbWl0ZXJsaW1pdDoxMDt9DQoJLnN0MXtmaWxsOm5vbmU7c3Ryb2tlOiMwMDAwMDA7c3Ryb2tlLXdpZHRoOjI7c3Ryb2tlLWxpbmVjYXA6cm91bmQ7c3Ryb2tlLWxpbmVqb2luOnJvdW5kO3N0cm9rZS1taXRlcmxpbWl0OjEwO30NCjwvc3R5bGU+DQo8cmVjdCB4PSI2IiB5PSIzIiBjbGFzcz0ic3QxIiB3aWR0aD0iMjAiIGhlaWdodD0iMjYiLz4NCjxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjE2IiBjeT0iMTkiIHI9IjciLz4NCjxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjE2IiBjeT0iMTkiIHI9IjIiLz4NCjxjaXJjbGUgY2xhc3M9InN0MSIgY3g9IjIxIiBjeT0iOCIgcj0iMiIvPg0KPC9zdmc+DQo=",
+  "name": "AudioByName",
+  "previewIconUrl": "https://resources.gdevelop-app.com/assets/Icons/Line Hero Pack/Master/SVG/Music/Music_speaker_audio_sound.svg",
+  "shortDescription": "Extension to allow playing audio by filename.",
+  "version": "1.0.0",
+  "description": "Allows for playing a sound resource by filename as a string. This also allows for dynamically selecting a filename via string expressions, which can be useful for things like playing voiced dialogue in larger games.",
+  "tags": [
+    "sound",
+    "music"
+  ],
+  "authorIds": [
+    "Khhq7a4Lu8bt91GLLhG0QiGLpfx2",
+    "8Ih1aa8f5gWUp4UB2BdhQ2iXWxJ3"
+  ],
+  "dependencies": [],
+  "eventsFunctions": [
+    {
+      "description": "Play Music on a Channel (By Name).",
+      "fullName": "Play Music on a Channel By Name",
+      "functionType": "Action",
+      "name": "PlayMusicOnChannelByName",
+      "sentence": "Plays _PARAM1_ on channel _PARAM2_ (Loop = _PARAM3_, Volume = _PARAM4_, Pitch = _PARAM5_)",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "const soundManager = runtimeScene._runtimeGame._soundManager;",
```

We cannot accept extensions that use private class properties, as these may break at any time! Please use the public API:
```suggestion
            "const soundManager = runtimeScene.getGame().getSoundManager();",
```

Same goes for all other actions in the extension.

---

These changes should now be completed.

---

@arthuro555 It has been forever since I did this, but I think you might have to mark these changes as reviewed before it can be merged now. Let me know if incorrect.